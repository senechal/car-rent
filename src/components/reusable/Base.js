import classNames from 'classnames';
import PropTypes from 'prop-types';

const Base = ({className, defaultCls, as: Tag, ...props}) => {
  return <Tag className={classNames(defaultCls, className)} {...props} />
};

Base.propTypes = {
  defaultCls: PropTypes.string.isRequired,
  as: PropTypes.string,
  className: PropTypes.string,
};

Base.defaultProps = {
  as: 'div',
  className: '',
};

export default Base;