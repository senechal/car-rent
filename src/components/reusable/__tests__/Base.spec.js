import { render } from '@testing-library/react';
import Base from '../Base';

describe('Testing Base', () => {
  it('should render correctly for a defaultClass', () => {
    const { container } = render(<Base defaultCls="base-class" />);
    expect(container).toMatchSnapshot();
  });
  it('should render correctly as a section', () => {
    const { container } = render(<Base defaultCls="base-class" as="section" />);
    expect(container).toMatchSnapshot();
  });
  it('should render correctly with an other className', () => {
    const { container } = render(<Base defaultCls="base-class" className="other-class" />);
    expect(container).toMatchSnapshot();
  });
});
