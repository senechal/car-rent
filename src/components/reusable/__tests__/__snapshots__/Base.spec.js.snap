// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Base should render correctly as a section 1`] = `
<div>
  <section
    class="base-class"
  />
</div>
`;

exports[`Testing Base should render correctly for a defaultClass 1`] = `
<div>
  <div
    class="base-class"
  />
</div>
`;

exports[`Testing Base should render correctly with an other className 1`] = `
<div>
  <div
    class="base-class other-class"
  />
</div>
`;
